libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/WebSocketConnector.cs

```diff
@@ -0,0 +1,148 @@
+
+                throw ex;
+            }
+
+            if (!httpRequest.HttpContext.WebSockets.IsWebSocketRequest)
+            {
+                httpRequest.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                await httpRequest.HttpContext.Response.WriteAsync("Upgrade to WebSocket is required.").ConfigureAwait(false);
+                _logger?.LogInformation("Invalid request: Request was not a WebSocket handshake.");
+
+                return;
+            }
+
+            try
+            {
+                if (!await _credentialProvider.IsAuthenticationDisabledAsync().ConfigureAwait(false))
+                {
+                    var authHeader = httpRequest.Headers.Where(x => x.Key.ToLower() == AuthHeaderName).FirstOrDefault().Value.FirstOrDefault();
+                    var channelId = httpRequest.Headers.Where(x => x.Key.ToLower() == ChannelIdHeaderName).FirstOrDefault().Value.FirstOrDefault();
+
+                    if (string.IsNullOrWhiteSpace(authHeader))
+                    {
+                        await MissingAuthHeaderHelperAsync(AuthHeaderName, httpRequest).ConfigureAwait(false);
+
+                        return;
+                    }
+
+                    if (string.IsNullOrWhiteSpace(channelId))
+                    {
+                        await MissingAuthHeaderHelperAsync(ChannelIdHeaderName, httpRequest).ConfigureAwait(false);
+
+                        return;
+                    }
+
+                    var claimsIdentity = await JwtTokenValidation.ValidateAuthHeader(authHeader, _credentialProvider, _channelProvider, channelId).ConfigureAwait(false);
+                    if (!claimsIdentity.IsAuthenticated)
+                    {
+                        httpRequest.HttpContext.Response.StatusCode = (int)HttpStatusCode.Unauthorized;
+                        _logger?.LogInformation("Unauthorized connection attempt.");
+
+                        return;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                httpRequest.HttpContext.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+                await httpRequest.HttpContext.Response.WriteAsync("Error while attempting to authorize connection.").ConfigureAwait(false);
+                _logger?.LogError(ex.Message);
+
+                return;
+            }
+
+            await CreateStreamingServerConnectionAsync(onTurnError, middlewareSet, httpRequest.HttpContext).ConfigureAwait(false);
+        }
+
+        private async Task MissingAuthHeaderHelperAsync(string headerName, HttpRequest httpRequest)
+        {
+            httpRequest.HttpContext.Response.StatusCode = (int)HttpStatusCode.Unauthorized;
+            _logger?.LogInformation($"Unable to authentiate. Missing header: {headerName}");
+            await httpRequest.HttpContext.Response.WriteAsync($"Unable to authentiate. Missing header: {headerName}").ConfigureAwait(false);
```

Comment:
I'm not sure why all this logging in here. ASP.Net already has this, and it seems redundant...